docs/rfcs/0007-flow-modules.md

```diff
@@ -0,0 +1,295 @@
+
+### Allow loading a module from a string
+
+Modules will not care about the source of a string. In the case of a `module.single` the module will take in one string of valid river configuration. In the case of `module.multiple` the module will take in a `map(string)` where the key uniquely denotes the module.
+
+### Allow modules to load other modules
+
+Modules will be able to load other modules, with reasonable safe guards. There will be a stack limit for the depth of sub-modules and modules will be unable to load themselves.
+
+### Modules should be sandboxed except via arguments and exports
+
+Modules cannot directly access children or parent modules except through predefined arguments and exports. 
+
+## Non Goals
+
+* Add additional capabilities to load strings
+* Any type of versioning
+* Any user interface work beyond ensuring it works as the UI currently does
+
+*Note* should security be a non-goal? For example can a submodule spawn a `local.file`, should there be the ability to pass a deny list of modules, and/or allow list? 
+
+*Note* should metadata be a non-goal? Metadata would allow better filtering the ability to tag modules, also will allow programmatic access to things that might otherwise be in comments. 
+
+### Add additional capabilities to load strings
+
+Modules will not care about the source of the string that represents the river syntax, nor will modules have any inherent reload semantics. The component supplying the string will be responsible for the source and will notify the module when the input changes so that it can utilize the new river configuration.
+
+### Any type of versioning
+
+Modules will not contain any sort of versioning nor will check for compatibility outside the normal river checks.
+
+### Any user interface work beyond ensuring it works as the UI currently does
+
+Users will not be able to drill into modules, they will be represented as any other normal component. 
+
+## Example
+
+```river
+// module
+argument "password" {
+    optional = false
+    comment = "password for mysql"
+}
+
+argument "username" {
+    optional = false
+    comment = "username for mysql"
+}
+
+export "targets" {
+    comment = "targets for the integration"
+    value = integrations.mysql.server1.targets
+}
+
+integrations.mysql "server1" {
+    username = argument.username.value
+    password = argument.password.value
+}
+
+```
+
+```river
+// parent
+
+local.file "mysql" {
+    filename = "/test/mysql.river"
+}
+
+module.single "mysql" {
+    content = local.file.mysql.content
+    arguments = {
+        {
+            "password" = PASSWORD,
+            "username" = USERNAME,
+        }
+    }
+}
+
+prometheus.scrape "scraper" {
+    targets = module.single.mysql.exports.targets
+}
+
+```
+
+## Limitations
+
+* Duplicate modules cannot be nested, this may or may not be enforced by the system
```

Comment:
Can we make a decision for this RFC whether module loaders prevent this?

Also, I'm not 100% sure what you mean by "duplicate modules cannot be nested." Do you mean that a "module cannot directly or indirectly load itself?" That may be clearer.

Reply:
Let's go with its not enforced by the system. Will change the verbiage.